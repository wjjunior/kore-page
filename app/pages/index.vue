<template>
  <div
    class="min-h-screen bg-gradient-to-br from-blue-50 to-indigo-100 flex items-center justify-center"
  >
    <div class="bg-white rounded-lg shadow-xl p-8 max-w-md w-full mx-4">
      <h1 class="text-3xl font-bold text-gray-800 mb-4 text-center">
        Homepage
      </h1>
    </div>
  </div>
</template>

<script setup lang="ts">
</script>

