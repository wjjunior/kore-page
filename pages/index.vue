<template>
  <HomePage />
</template>

<script setup lang="ts">
import HomePage from "@/components/home";
</script> 