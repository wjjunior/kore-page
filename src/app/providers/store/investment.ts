import { defineStore } from "pinia";
import type { InvestmentBannerData } from "@/shared/lib/types";
import { investmentAPI } from "@/shared/lib/api";

interface InvestmentState {
  bannerData: InvestmentBannerData | null;
  loading: boolean;
  error: string | null;
}

export const useInvestmentStore = defineStore("investment", {
  state: (): InvestmentState => ({
    bannerData: null,
    loading: false,
    error: null,
  }),

  getters: {
    isDataLoaded: (state) => state.bannerData !== null,
    getDaysLeft: (state) => state.bannerData?.daysLeft ?? 0,
    getTotalInvestors: (state) => state.bannerData?.totalInvestors ?? 0,
    getFundingGoal: (state) => state.bannerData?.fundingGoal ?? 0,
    getFundsRaised: (state) => state.bannerData?.fundsRaised ?? 0,
    getMinimumInvestment: (state) => state.bannerData?.minimumInvestment ?? 0,
    getDeadline: (state) => state.bannerData?.deadline ?? "",
    getTypeOfSecurity: (state) => state.bannerData?.typeOfSecurity ?? "",
    getRevenueShareDuration: (state) =>
      state.bannerData?.revenueShareDuration ?? 0,
    getCategories: (state) => state.bannerData?.categories ?? [],
    getCompanyName: (state) => state.bannerData?.companyName ?? "",
    getCompanyDescription: (state) =>
      state.bannerData?.companyDescription ?? "",
    getWebsite: (state) => state.bannerData?.website ?? "",
    getOfferingTerms: (state) => state.bannerData?.offeringTerms ?? [],
    getDocuments: (state) => state.bannerData?.documents ?? [],
    getTeamMembers: (state) => state.bannerData?.teamMembers ?? [],
    getTeamDescription: (state) => state.bannerData?.teamDescription ?? "",
    getMarketingPlan: (state) => state.bannerData?.marketingPlan ?? "",
    getFaqItems: (state) => state.bannerData?.faqItems ?? [],
  },

  actions: {
    async fetchInvestmentsData() {
      this.loading = true;
      this.error = null;

      try {
        const data = await investmentAPI.fetchInvestmentsData();
        this.bannerData = data;
      } catch (error) {
        this.error =
          error instanceof Error
            ? error.message
            : "Failed to fetch investment data";
        console.error("Error fetching banner data:", error);
      } finally {
        this.loading = false;
      }
    },

    clearError() {
      this.error = null;
    },
  },
});
