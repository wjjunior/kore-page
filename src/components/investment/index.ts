export { default } from "./ui/InvestmentPage.vue";
export { default as InvestmentOverview } from "../../features/investment/ui/InvestmentOverview.vue";
