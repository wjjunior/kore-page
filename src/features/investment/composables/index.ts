export { useInvestmentData } from "./useInvestmentData";
export { useInvestmentDataSSR } from "./useInvestmentDataSSR";
