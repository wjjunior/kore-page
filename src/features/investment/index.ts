export * from "./ui";
export * from "./composables";
export * from "./lib";
