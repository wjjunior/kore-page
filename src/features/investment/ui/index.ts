export { default as InvestmentOverview } from "./InvestmentOverview.vue";
export { default as InvestmentDetails } from "./InvestmentDetails.vue";
export { default as InvestmentCards } from "./InvestmentCards.vue";
export { default as InvestmentTeam } from "./InvestmentTeam.vue";
export { default as InvestmentMarketingPlan } from "./InvestmentMarketingPlan.vue";
export { default as InvestmentFaq } from "./InvestmentFaq.vue";
