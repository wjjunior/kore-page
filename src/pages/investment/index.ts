export { default as InvestmentPage } from "./ui/InvestmentPage.vue";
