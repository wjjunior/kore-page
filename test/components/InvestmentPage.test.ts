import { describe, it, expect, vi, beforeEach } from "vitest";
import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import { InvestmentPage } from "@/pages/investment";
import { mockInvestmentBannerData } from "../mocks";

vi.mock("@/features/investment", async () => {
  const actual = await vi.importActual<typeof import("@/features/investment")>(
    "@/features/investment"
  );
  return {
    ...actual,
    InvestmentOverview: {
      name: "InvestmentOverview",
      template:
        '<div class="investment-overview-mock">Investment Overview Component</div>',
    },
    InvestmentDetails: {
      name: "InvestmentDetails",
      template:
        '<div class="investment-details-mock">Investment Details Component</div>',
    },
    useInvestmentPageController: () => ({
      loading: { value: false },
      errorMessage: "",
      isDataReady: { value: true },
      investmentData: {
        value: mockInvestmentBannerData,
      },
      refresh: vi.fn(),
    }),
  };
});

vi.mock("@/features/investment/composables/useInvestmentDataSSR", () => ({
  useInvestmentDataSSR: () => ({
    loading: { value: false },
    error: { value: null },
    isDataReady: { value: true },
    refresh: vi.fn(),
    investmentInfo: {
      value: mockInvestmentBannerData,
    },
  }),
}));

vi.mock("@/widgets/header", () => ({
  Header: {
    name: "Header",
    template: '<header class="header-mock">Header Component</header>',
  },
}));

vi.mock("#app", () => ({
  ClientOnly: {
    name: "ClientOnly",
    template: "<div><slot /></div>",
  },
}));

const mountInvestmentPage = () => {
  return mount(InvestmentPage, {
    global: {
      stubs: {
        ClientOnly: { template: "<div><slot /></div>" },
        Footer: { template: '<footer class="footer-mock">Footer</footer>' },
        LoadingSpinner: {
          template: '<div class="loading-spinner-mock">Loading...</div>',
        },
      },
    },
  });
};

describe("InvestmentPage", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it("renders header and footer", () => {
    const wrapper = mountInvestmentPage();
    expect(wrapper.find(".header-mock").exists()).toBe(true);
    expect(wrapper.find(".footer-mock").exists()).toBe(true);
  });

  it("renders investment overview and details components when data is ready", () => {
    const wrapper = mountInvestmentPage();
    expect(wrapper.find(".investment-overview-mock").exists()).toBe(true);
    expect(wrapper.find(".investment-details-mock").exists()).toBe(true);
  });

  it("matches snapshot when data is ready", () => {
    const wrapper = mountInvestmentPage();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
