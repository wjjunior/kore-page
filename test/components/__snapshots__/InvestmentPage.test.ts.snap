// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InvestmentPage > matches snapshot when data is ready 1`] = `
"<div>
  <header class="header-mock" active-nav-item="invest">Header Component</header>
  <div>
    <div class="investment-overview-mock" investment-data="[object Object]" loading="[object Object]" error="">Investment Overview Component</div>
    <div fallback-tag="div" fallback-class="min-h-[400px] flex items-center justify-center">
      <div class="investment-details-mock" investment-data="[object Object]" loading="[object Object]">Investment Details Component</div>
    </div>
  </div>
  <footer class="footer-mock">Footer</footer>
</div>"
`;
